ctapipe/io/hdf5merger.py

```diff
@@ -0,0 +1,470 @@
+    ).tag(config=True)
+
+    simulation = traits.Bool(
+        True,
+        help="Whether to include data only known for simulations in merged output",
+    ).tag(config=True)
+
+    true_images = traits.Bool(
+        True,
+        help="Whether to include true images in merged output",
+    ).tag(config=True)
+
+    true_parameters = traits.Bool(
+        True,
+        help="Whether to include parameters calculated on true images in merged output",
+    ).tag(config=True)
+
+    dl1_images = traits.Bool(
+        True,
+        help="Whether to include dl1 images in merged output",
+    ).tag(config=True)
+
+    dl1_parameters = traits.Bool(
+        True,
+        help="Whether to include dl1 image parameters in merged output",
+    ).tag(config=True)
+
+    dl2_subarray = traits.Bool(
+        True, help="Whether to include dl2 subarray-event-wise data in merged output"
+    ).tag(config=True)
+
+    dl2_telescope = traits.Bool(
+        True, help="Whether to include dl2 telescope-event-wise data in merged output"
+    ).tag(config=True)
+
+    monitoring = traits.Bool(
+        True, help="Whether to include monitoring data in merged output"
+    ).tag(config=True)
+
+    processing_statistics = traits.Bool(
+        True, help="Whether to include processing statistics in merged output"
+    ).tag(config=True)
+
+    def __init__(self, output_path=None, **kwargs):
+        # enable using output_path as posarg
+        if output_path not in {None, traits.Undefined}:
+            kwargs["output_path"] = output_path
+
+        super().__init__(**kwargs)
+
+        if self.overwrite and self.append:
+            raise traits.TraitError("overwrite and append are mutually exclusive")
+
+        output_exists = self.output_path.exists()
+        appending = False
+        if output_exists and not (self.append or self.overwrite):
+            raise traits.TraitError(
+                f"output_path '{self.output_path}' exists but neither append nor overwrite allowed"
+            )
+
+        if output_exists and self.append:
+            appending = True
+
+        self.h5file = tables.open_file(
+            self.output_path,
+            mode="a" if appending else "w",
+            filters=DEFAULT_FILTERS,
+        )
+        Provenance().add_output_file(str(self.output_path))
+
+        self.required_nodes = set()
+        self.data_model_version = None
+        self.subarray = None
+        self.meta = None
+        # output file existed, so read subarray and data model version to make sure
+        # any file given matches what we already have
+        if appending:
+            try:
+                self.meta = metadata.Reference.from_dict(
+                    metadata.read_metadata(self.h5file)
+                )
+            except Exception:
+                raise CannotMerge(
+                    f"CTA Rerence meta not found in existing output file: {self.output_path}"
```

Comment:
could you use self._read_meta() here?

Reply:
Yes, was introduced later and forgot to use everywhere